crates/stc_ts_file_analyzer/src/analyzer/scope/vars.rs

```diff
@@ -120,7 +120,17 @@ impl Analyzer<'_, '_> {
                     }
                 }
 
-                let mut ty = opt_union(span, ty, default);
+                let mut ty = match (default, ty) {
+                    (Some(default), Some(ty)) => {
+                        if self.extends(span, &default, &ty, Default::default()).is_some() {
```

Comment:
I think `extends().is_some()` is wrong because `extends()` returns `None` only if it cannot be determined. i.e., type parameter

Reply:
I think it should be `if let Some(true) = self.extends(...) {} else {}`

Reply:
Oops, gotcha, fixed!